src/main/java/spoon/support/util/internal/ElementNameMap.java

```diff
@@ -33,22 +40,42 @@
  * <ul>
  *   <li>each inserted {@link CtElement} gets assigned correct parent</li>
  *   <li> each change is reported in {@link FineModelChangeListener}</li>
+ *   <li>The {@link ElementNameMap#entrySet()} method returns elements in the order they were inserted</li>
  * </ul>
  * <br>
  */
 public abstract class ElementNameMap<T extends CtElement> extends AbstractMap<String, T>
 		implements Serializable {
 
-	private static final long serialVersionUID = 1L;
+	private static final long serialVersionUID = 2L;
 
 	// This can not be a "Map" as this class is Serializable and therefore Sorald wants this field
 	// to be serializable as well (Rule 1948).
 	// It doesn't seem smart enough to realize it is final and only assigned to a Serializable Map
 	// in the constructor.
-	private final ConcurrentSkipListMap<String, T> map;
+	private final ConcurrentHashMap<String, InsertOrderWrapper<T>> map;
```

Comment:
The hash map generally performs better. The ordering of the skip list map wasn't what we wanted.